pystac_client/item_search.py

```diff
@@ -451,6 +453,7 @@ def get_items(self) -> Iterator[Item]:
                 if self._max_items and nitems >= self._max_items:
                     return
 
+    @lru_cache()
```

Comment:
one concern I have here is that the results could be tens of MB, so with the default cache size of 128, these could have GBs of cache. I don't know if these ever update or not, but setting max_size to 1 might be right if these never change?

Reply:
Makes sense to me.